### PR TITLE
fix: path error in Windows

### PR DESCRIPTION
**Pull request by DevHaoZi**
Closes goravel/goravel#181

## 📑 Description
Fix path error in Windows and optimize code.

## ✅ Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

## ℹ Additional Information
none.

### CODE REVIEW

File: filesystem/file.go [hunk 1]
```diff
@@ -44,14 +43,24 @@ func NewFileFromRequest(fileHeader *multipart.FileHeader) (*File, error) {
...
 	}
-	defer src.Close()
-
-	tempFileName := fmt.Sprintf("%s_*%s", ConfigFacade.GetString("app.name"), path.Ext(fileHeader.Filename))
-	tempFile, err := ioutil.TempFile(os.TempDir(), tempFileName)
+	defer func(src multipart.File) {
+		err = src.Close()
+		if err != nil {
+			panic(err)
+		}
```
Comment (hwbrzzl) [inline filesystem/file.go:50]:
```suggestion
		if err := src.Close(); err != nil {
			panic(err)
		}
```

File: filesystem/file.go [hunk 2]
```diff
@@ -44,14 +43,24 @@ func NewFileFromRequest(fileHeader *multipart.FileHeader) (*File, error) {
...
+	tempFile, err := os.CreateTemp(os.TempDir(), tempFileName)
 	if err != nil {
 		return nil, err
 	}
-	defer tempFile.Close()
+	defer func(tempFile *os.File) {
+		err = tempFile.Close()
+		if err != nil {
+			panic(err)
+		}
```
Comment (hwbrzzl) [inline filesystem/file.go:62]:
```suggestion
		if err := tempFile.Close(); err != nil {
			panic(err)
		}
```

**Comment by hwbrzzl** [comment]
CI failed

**Comment by DevHaoZi** [comment]
> CI failed

fixed

**Comment by hwbrzzl** [comment]
Oh, I forgot that this PR should be merged into v1.12.x, I'll do it.

**Comment by DevHaoZi** [comment]
> Oh, I forgot that this PR should be merged into v1.12.x, I'll do it.

ok, thanks.

**Pull request merged**